geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/geotools/SimpleFeatureTypes.scala

```diff
@@ -447,6 +447,59 @@ object SimpleFeatureTypes {
     renamed
   }
 
+  /**
+    * Compare two feature types. This method compares the the schemas of each feature type, i.e. the
+    * names and type bindings of the attributes. The result will be:
+    *
+    * <ul>
+    *   <li>a positive int if type A is a sub type or reorder of type B</li>
+    *   <li>zero if type A has the same schema as type B</li>
+    *   <li>a negative int if none of the above</li>
+    * </ul>
+    *
+    * Note that this shouldn't be used for sorting.
+    *
+    * Note: GeoTools DataUtilities has a similar method that does not work correctly
```

Comment:
How hard would it be to submit a PR for a fix to GeoTools?

Reply:
I think the main issue is this [line](https://github.com/geotools/geotools/blob/master/modules/library/main/src/main/java/org/geotools/data/DataUtilities.java#L472) should be `a.getType.getBinding.equals(b.getType.getBinding)` - I'm somewhat hesitant to change the behavior though... What do you think?

Reply:
Hmmm....  if it is 'broken' as-is, then there ought to be a unit test which would show it.   

If it isn't a major pain to gin up a test and patch, then we could discuss it with the community in a PR. 

That said, DataUtilities is considered 'sample' code; despite that, (I think) it is used widely.  Those two considerations are at odds with each other, so GeoTools may or may not merge the fix.

Reply:
I'll put up a pr and we can see.

Reply:
https://github.com/geotools/geotools/pull/2525